drivers/timer/altera_avalon_timer_hal.c

```diff
@@ -69,3 +74,15 @@ uint32_t sys_clock_cycle_get_32(void)
 	 */
 	return accumulated_cycle_count;
 }
+
+void wrapped_announce(int32_t ticks)
+{
+	driver_uptime += ticks;
+	sys_clock_announce(ticks);
+}
+#define sys_clock_announce(t) wrapped_announce(t)
```

Comment:
This `#define` seems to be misplaced.

Since `sys_clock_announce` is not overridable (i.e. not a weak symbol), I would say the best way to handle this would be to explicitly use `wrapped_announce` throughout this file.